internals/daemon/daemon_test.go

```diff
@@ -340,153 +341,140 @@ func (s *daemonSuite) TestFillsWarnings(c *C) {
 func (s *daemonSuite) TestGuestAccess(c *C) {
 	d := s.newDaemon(c)
 
-	get := &http.Request{Method: "GET"}
-	put := &http.Request{Method: "PUT"}
-	pst := &http.Request{Method: "POST"}
-	del := &http.Request{Method: "DELETE"}
-
-	cmd := &Command{d: d}
-	c.Check(cmd.canAccess(get, nil), Equals, accessUnauthorized)
```

Comment:
Since we can't test `canAccess()` directly anymore, we instead indirectly test the access checkers and how they relate to `GET` / `PUT` / `POST`.

Conceptually it's the same:

- Creation of `Command` with `AdminOnly`, `UserOK`, `GuestOK` is replaced with `ReadAccess` and `WriteAccess`
- Calling `canAccess()` with different methods is replaced with calling `doTestReqFunc`, which calls through `cmd.ServeHTTP`, and checks the returned HTTP status